daemon/compartment.c

```diff
@@ -1203,8 +1326,8 @@ compartment_start_post_clone_early_cb(int fd, unsigned events, event_io_t *io, v
 	char msg_stop = COMPARTMENT_START_SYNC_MSG_STOP;
 	if (write(compartment->sync_sock_parent, &msg_stop, 1) < 0) {
 		WARN_ERRNO("write to sync socket failed");
+		compartment_kill(compartment);
 	}
-	compartment_kill(compartment);
```

Comment:
is the compartment boot timeout registered before this is reached?

Reply:
does this matter. cleanup should be triggered in sigchld_handler, which derigisters it if set and otherwise
ignore.